frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala

```diff
@@ -1,81 +1,86 @@
 
-import scala.concurrent.duration.{FiniteDuration, _}
 import scala.scalajs.js
-import scala.scalajs.js.timers.SetTimeoutHandle
+import scala.scalajs.js.JSConverters._
+import scala.scalajs.js.|
+import scala.scalajs.js.|._
 
 class GraphView(showUnconnected: Var[Boolean]) extends Component {
 
-  override def render: Element = {
-
-    //FIXME: port initialization of graph from JS to scala and do it here
-    js.undefined.asInstanceOf
+  private[this] lazy val visibleNodes = js.Dictionary[Unit]()
+  private[this] lazy val networkNodes = new vis.DataSet[vis.Node]()
+  private[this] lazy val networkEdges = new vis.DataSet[vis.Edge]()
+  private[this] var network: js.UndefOr[vis.Network] = js.undefined
+
+  override def attach(parent: Element): Unit = {
+    console.log(networkNodes)
+    console.log(networkEdges)
+    console.log(parent)
+    val data = vis.NetworkData(networkNodes, networkEdges)
+    console.log(data)
+    network.foreach(_.destroy())
+    network = new vis.Network(parent, data, vis.NetworkOptions())
+    console.log(network)
   }
 
-  private val graph = DOMGlobalScope.graph
-  private val nodes = js.Dictionary[js.Dictionary[js.Any]]()
-  private val connectedNodes = js.Dictionary[Unit]()
-  private val createdLinks = js.Dictionary[Unit]()
-
-  private val updateFrequency: FiniteDuration = 500.millis
-
-  private var scheduler: js.UndefOr[SetTimeoutHandle] = js.undefined
+  private[this] val scheduler = new ScheduledQueue[GraphView.GraphOperation](applyGraphOperations)
+  private[this] val nodeData = js.Dictionary[String]()
+  private[this] val connectedNodes = js.Dictionary[Unit]()
+  private[this] val createdLinks = js.Dictionary[Unit]()
+  private[this] lazy val fitOnce = network.foreach(_.fit) // do it once
 
-  private def isNodeConnected(node: String): Boolean = {
+  private[this] def isNodeConnected(node: String): Boolean = {
     connectedNodes.contains(node)
   }
 
   showUnconnected.triggerLater {
     val show = showUnconnected.now
     if (show) {
-      nodes.foreach {
-        case (node, data) =>
-          enqueueOperation(GraphView.AddNode(node, data))
+      nodeData.foreach {
+        case (node, label) =>
+          scheduler.enqueueOperation(GraphView.AddNode(node, label))
       }
     } else {
-      nodes.foreach {
-        case (node, data) if !isNodeConnected(node) =>
-          enqueueOperation(GraphView.RemoveNode(node))
+      nodeData.foreach {
+        case (node, label) if !isNodeConnected(node) =>
+          scheduler.enqueueOperation(GraphView.RemoveNode(node))
         case _ => //do nothing
       }
     }
   }
 
-  private def scheduleGraphOperations(): Unit = {
-    if (scheduler.isEmpty) {
-      val timer: SetTimeoutHandle = scala.scalajs.js.timers.setTimeout(updateFrequency) {
-        scheduler = js.undefined
-        applyGraphOperations()
-      }
-      scheduler = timer
-    }
-  }
+  private def applyGraphOperations(operationsToApply: js.Array[GraphView.GraphOperation]): Unit = {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [applyGraphOperations should have a smaller scope by using private[this]](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703402/issues/source?bid=3121282&fileBranchId=3158877#l56)